packages/next/src/build/index.ts

```diff
@@ -2956,15 +2958,22 @@ export default async function build(
             const file = normalizePagePath(page)
 
             const pageInfo = pageInfos.get(page)
-            const durationInfo = exportConfig.pageDurationMap[page]
+            const durationInfo = exportResult.byPage.get(page)
             if (pageInfo && durationInfo) {
               // Set Build Duration
               if (pageInfo.ssgPageRoutes) {
                 pageInfo.ssgPageDurations = pageInfo.ssgPageRoutes.map(
-                  (pagePath) => durationInfo[pagePath]
+                  (pagePath) => {
+                    const duration = durationInfo.durationsByPath.get(pagePath)
+                    if (typeof duration === 'undefined') {
```

Comment:
There were a few situations that I noticed that the pathname data was undefined, so I added some runtime checks. Ideally, to solve this once-and-for-all, we should leverage the return data from the `exportApp` to iterate over instead such that the export can return the files and data it generated or created so we don't have to _guess_ and _infer_ here.